src/HttpMultipart/Message/MultipartResponse.php

```diff
@@ -57,10 +57,15 @@ public static function parseMultipartBody(StreamInterface $stream)
           break;
         }
         $headers = [];
-        list($header_lines, $body) = explode("\r\n\r\n", $message, 2);
+        $message_parts = explode("\r\n\r\n", $message, 2);
+        $header_lines = $message_parts[0];
+        $body = isset($message_parts[1])? $message_parts[1] : NULL;
         foreach (explode("\r\n", $header_lines) as $header_line) {
-          list($key, $value) = preg_split('/:\s+/', $header_line, 2);
-          $headers[strtolower($key)] = $value;
+          $header_parts = preg_split('/:\s+/', $header_line, 2);
+          if (count($header_parts) == 2) {
+            list($key, $value) = $header_parts;
+            $headers[strtolower($key)] = $value;
+          }
```

Comment:
Let's add some comments to what's going on. It's some pretty complicated logic here :smile: